Source/Core/DolphinQt/Achievements/AchievementHeaderWidget.cpp

```diff
@@ -0,0 +1,140 @@
+  m_user_points = new QLabel();
+  m_game_name = new QLabel();
+  m_game_points = new QLabel();
+  m_game_progress_hard = new QProgressBar();
+  m_game_progress_soft = new QProgressBar();
+  m_rich_presence = new QLabel();
+
+  QVBoxLayout* m_user_right_col = new QVBoxLayout();
+  m_user_right_col->addWidget(m_user_name);
+  m_user_right_col->addWidget(m_user_points);
+  QHBoxLayout* m_user_layout = new QHBoxLayout();
+  // TODO lillyjade 02 Jun 2023 - player badge goes here
+  m_user_layout->addLayout(m_user_right_col);
+  m_user_box = new QGroupBox();
+  m_user_box->setLayout(m_user_layout);
+
+  QVBoxLayout* m_game_right_col = new QVBoxLayout();
+  m_game_right_col->addWidget(m_game_name);
+  m_game_right_col->addWidget(m_game_points);
+  m_game_right_col->addWidget(m_game_progress_hard);
+  m_game_right_col->addWidget(m_game_progress_soft);
+  QHBoxLayout* m_game_upper_row = new QHBoxLayout();
+  // TODO lillyjade 02 Jun 2023 - player badge and game badge go here
+  m_game_upper_row->addLayout(m_game_right_col);
+  QVBoxLayout* m_game_layout = new QVBoxLayout();
+  m_game_layout->addLayout(m_game_upper_row);
+  m_game_layout->addWidget(m_rich_presence);
+  m_game_box = new QGroupBox();
+  m_game_box->setLayout(m_game_layout);
+
+  QVBoxLayout* m_total = new QVBoxLayout();
+  m_total->addWidget(m_user_box);
+  m_total->addWidget(m_game_box);
+
+  UpdateData();
+
+  m_total->setContentsMargins(0, 0, 0, 0);
+  m_total->setAlignment(Qt::AlignTop);
+  setLayout(m_total);
+}
+
+void AchievementHeaderWidget::UpdateData()
+{
+  if (!AchievementManager::GetInstance()->IsLoggedIn())
+  {
+    m_user_box->setVisible(false);
+    m_game_box->setVisible(false);
+    return;
+  }
+
+  QString user_name =
+      QString::fromStdString(AchievementManager::GetInstance()->GetPlayerDisplayName());
+  m_user_name->setText(user_name);
+  m_user_points->setText(tr("%1 points").arg(AchievementManager::GetInstance()->GetPlayerScore()));
+
+  if (!AchievementManager::GetInstance()->IsGameLoaded())
+  {
+    m_user_box->setVisible(true);
+    m_game_box->setVisible(false);
+    return;
+  }
+
+  AchievementManager::PointSpread point_spread = AchievementManager::GetInstance()->TallyScore();
+  m_game_name->setText(
+      QString::fromStdString(AchievementManager::GetInstance()->GetGameDisplayName()));
+  m_game_points->setText(GetPointsString(user_name, point_spread));
+  m_game_progress_hard = new QProgressBar();
+  m_game_progress_hard->setRange(0, point_spread.total_count);
+  m_game_progress_soft->setValue(point_spread.hard_unlocks);
+  m_game_progress_soft->setRange(0, point_spread.total_count);
+  m_game_progress_soft->setValue(point_spread.hard_unlocks + point_spread.soft_unlocks);
+  // TODO lillyjade 02 Jun 2023 - once RP is working again, add it here
+  // m_rich_presence->setText(QString::fromStdString(AchievementManager::GetInstance()->GenerateRichPresence()));
+  // m_rich_presence->setVisible(Config::Get(Config::RA_RICH_PRESENCE_ENABLED));
```

Comment:
What's preventing this from working right now?

Reply:
Originally when I created this PR, you had identified an outstanding bug with the RP that I put in a separate PR. When I went back to look at this later, I realized however that I had no good way of easily exposing the RP the way it was currently structured. I have a commit on a separate branch that refactors the RP to be useful here and resolves this comment, I was going to put it in a separate PR to keep this one clean.